src/transcode_image.rs

```diff
@@ -0,0 +1,114 @@
+use actix_web::{Either, HttpResponse, HttpResponseBuilder};
+use image::ImageError;
+use once_cell::sync::Lazy;
+use reqwest::Response;
+use tokio::task::spawn_blocking;
+
+pub static DISALLOW_IMAGE_TRANSCODING: Lazy<bool> =
+    Lazy::new(|| crate::utils::get_env_bool("DISALLOW_IMAGE_TRANSCODING"));
+
+#[derive(Debug, thiserror::Error)]
+enum ImageTranscodingError {
+    #[error("Image loading error: {0}")]
+    ImageLoadingError(#[from] ImageError),
+    #[cfg(feature = "webp")]
+    #[error("Image is not an 8bit RGB")]
+    NotAnRgb8Image,
+}
+
+pub async fn transcode_image(
+    image_response: Response,
+    http_response: &mut HttpResponseBuilder,
+    #[cfg(feature = "avif")] avif: bool,
+) -> Result<Either<HttpResponse, Response>, Box<dyn std::error::Error>> {
+    let Some(content_type) = image_response.headers().get("content-type") else {
+        return Ok(Either::Right(image_response));
+    };
+
+    #[cfg(feature = "avif")]
+    if content_type == "image/webp" || content_type == "image/jpeg" && avif {
```

Comment:
Isn't this check supposed to be like that?
```suggestion
    if (content_type == "image/webp" || content_type == "image/jpeg") && avif {
```
Now `image/webp` is always transformed regardless of the presence of `avif` in the query, because this check is computed as
```rs
if content_type == "image/webp" || (content_type == "image/jpeg" && avif) {
```